src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -148,14 +148,15 @@ public function onSubmit(FormEvent $event)
             throw new UnexpectedTypeException($data, 'array or (\Traversable and \ArrayAccess)');
         }
 
-        if ($this->deleteEmpty) {
-            $previousData = $event->getForm()->getData();
+        if ($entryFilter = $this->deleteEmpty) {
```

Comment:
Yep, but why not use `$this->deleteEmpty` directly? IMHO `$this->deleteEmpty($child->getData())` is quite easy to understand too.